### ISSUE TITLE
Improve support for Bitbucket

### ISSUE THREAD

**Issue by alikhajeh1** [infracost/infracost#1173]
There are three things we should do:
- [x] create a summary table format for Bitbucket comments. This would be similar to what we do for GitHub but since Bitbucket doesn't support HTML yet, we could use their markdown format to do something similar. I'm not sure if Bitbucket has comment edit histories like GitHub does but that's not something we can change anyway.
        ![summary-table](https://user-images.githubusercontent.com/638577/144510622-0886cb20-32e9-4c26-b324-96ebd5583327.png)
- [x] add `infracost comment bitbucket` that supports cloud and self-hosted Bitbucket versions 
- [x] add a `bitbucket-comment` format to `infracost output`

## Note for community
Please 👍 if you'd like to upvote this or receive updates.

**Comment by wahlfeld10x** [comment]
Would be good to include support for Bitbucket Server/Data Center as well, as from what I can tell, the only difference is we need to use a Bitbucket API key instead of an app password. I've gotten this to work with `diff.sh` but wouldn't mind being kept in the loop on how support with `compost` is going.

**Comment by alikhajeh1** [comment]
@wahlfeld10x sounds good! Can you possibly share the curl/bash snippet you used to make it work for Bitbucket Server/Data Center? That helps a lot when we come to add it to compost :)

**Comment by wahlfeld10x** [comment]
Sure here you go, this all went in `diff.sh`. Tested with Atlassian Bitbucket v7.17.1.

```bash
# For Bitbucket Server: BITBUCKET_SERVER_HOSTNAME must be set, e.g. "mycompany.com". BITBUCKET_TOKEN must be set to "mytoken", the token
#   needs to have PROJECT READ and REPOSITORY READ on "Repositories" and "Pull Requests" so it can post comments. See here for my details:
#   https://confluence.atlassian.com/bitbucketserver/personal-access-tokens-939515499.html

get_bitbucket_server_url () {
  BITBUCKET_GIT_URL="$(git config --get remote.origin.url)"
  BITBUCKET_PROJECT="$(cut -d '/' -f4 <<<$BITBUCKET_GIT_URL)"
  BITBUCKET_REPO="$(basename $BITBUCKET_GIT_URL .git)"
  printf "https://$BITBUCKET_SERVER_HOSTNAME/rest/api/1.0/projects/$BITBUCKET_PROJECT/repos/$BITBUCKET_REPO/pull-requests"
}

get_bitbucket_server_pr_id () {
  BITBUCKET_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
  curl -sS \
    -H "Authorization: Bearer $BITBUCKET_TOKEN" \
    "$1?at=refs/heads/$BITBUCKET_BRANCH&direction=OUTGOING" \
    | jq -r '.["values"][0].id'
}

post_bitbucket_server_comment () {
  msg="$(build_msg false)"
  echo "Posting comment to $1"
  jq -Mnc --arg msg "$msg" '{"text": "\($msg)"}' | curl -sSL -o /dev/null -X POST -d @- \
    -H "Content-Type: application/json" \
    -H "Authorization: Bearer $BITBUCKET_TOKEN" \
    "$1"
}

post_to_bitbucket_server () {
  if [ -z "$BITBUCKET_TOKEN" ]; then
    echo "Error: BITBUCKET_TOKEN is required to post comment to Bitbucket Server"
  else
    BITBUCKET_SERVER_URL="$(get_bitbucket_server_url)"
    BITBUCKET_SERVER_PR_ID="$(get_bitbucket_server_pr_id $BITBUCKET_SERVER_URL)"
    post_bitbucket_server_comment "$BITBUCKET_SERVER_URL/$BITBUCKET_SERVER_PR_ID/comments"
  fi
}

elif [ -n "$BITBUCKET_SERVER_HOSTNAME" ]; then
  post_to_bitbucket_server
```

**Comment by alikhajeh1** [comment]
This is [in progress](https://github.com/infracost/infracost/pull/1363#discussion_r803916399) and will be shipped early next week in v0.9.18 🚀 

The [docs PR](https://github.com/infracost/docs/pull/164/files) shows what the command will look like, it'll support the Cloud and Bitbucket Server/Data Center versions

**Comment by vdmgolub** [comment]
This is live in v0.9.18.
